packages/dtc/1.4.7/0001-enable-install-exe.patch

```diff
@@ -0,0 +1,31 @@
+From 259dc70b4b3bcfa2457650d7e07eba608b748cb1 Mon Sep 17 00:00:00 2001
+From: toriningen <toriningen.me@gmail.com>
+Date: Sun, 22 Dec 2024 16:07:09 +0200
+Subject: [PATCH] Enable install of .exe
+
+---
+ Makefile | 8 +++++++-
+ 1 file changed, 7 insertions(+), 1 deletion(-)
+
+diff --git a/Makefile b/Makefile
+index d8ebc4f..3c4e4a6 100644
+--- a/Makefile
++++ b/Makefile
+@@ -199,7 +199,13 @@ endif
+ install-bin: all $(SCRIPTS)
+ 	@$(VECHO) INSTALL-BIN
+ 	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+-	$(INSTALL_PROGRAM) $(BIN) $(DESTDIR)$(BINDIR)
++	for bin in $(BIN); do \
++	  if [ -f "$$bin" ]; then \
++	    $(INSTALL_PROGRAM) $$bin $(DESTDIR)$(BINDIR); \
++	  elif [ -f "$$bin.exe" ]; then \
```

Comment:
The way this is usually done would involve having a variable (e.g. `EXEEXT`) that is defined to nothing on Linux/BSD/etc and defined to `.exe` on msys/windows. Regardless see what you can land upstream first then ct-ng can carry a backport.

Reply:
> The way this is usually done would involve having a variable (e.g. `EXEEXT`) that is defined to nothing on Linux/BSD/etc and defined to `.exe` on msys/windows. Regardless see what you can land upstream first then ct-ng can carry a backport.

That's the route I wanted to go, but the challenge is that technically this is happening on Linux, we are cross-compiling for Windows, so uname won't match the extension.

I suppose it's configure's job to detect EXEEXT, but I can in theory do this in Makefile myself with something like this?

```
EXEEXT := $(shell echo | "$(CC)" -x c - '-###' -o foobar 2>&1 | \
	    sed -En '/-o foobar[^ ]*/s/.*-o foobar([^ ]*).*?/\1/p' | \
	    head -1)
```